runtime/common/ExecutionContext.h

```diff
@@ -127,5 +127,14 @@ class ExecutionContext {
   /// @brief Whether or not to simply concatenate measurements in execution
   /// order.
   bool explicitMeasurements = false;
+
+  /// @brief Counts of resources (gates and qubits) used when in resource
+  /// counting mode
+  resource_counts resourceCounts;
```

Comment:
can `resourceCounts` and `choice` function be part of the resource count simulator? Is there a reason to save them on the execution context?

Reply:
We've figured out a nice solution that negates this need.